Feature branch: Transactions and alerts

alonp99 opened the pull request:
### Description
Elaborate on the subject, motivation, and context.

### Related issues
 * Provide a link to each related issue.

### Breaking changes
 * Describe the breaking changes that this pull request introduces.

### How these changes were tested
 * Describe the tests that you ran to verify your changes, including devices, operating systems, browsers and versions.

### Examples and references
 * Links, screenshots, and other resources related to this change.

### Checklist
- [] I have read the [contribution guidelines](CONTRIBUTING.md) of this project
- [] I have read the [style guidelines](STYLE_GUIDE.md) of this project
- [] I have performed a self-review of my own code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings and errors
- [] New and existing tests pass locally with my changes

alonp99 committed:
feat(wf-service): transaction modules

alonp99 committed:
fix(docker-compose): fix compose

alonp99 committed:
feat(wf-service): transaction models

alonp99 committed:
feat(wf-service): transaction models

alonp99 committed:
feat(wf-service): transaction models

alonp99 committed:
merged with master

alonp99 committed:
merged with master

liorzblrn committed:
fix: fix build

liorzblrn committed:
fix: fix build

alonp99 committed:
merged with master

alonp99 committed:
merged with master

alonp99 committed:
merged with master

alonp99 committed:
merged with master

alonp99 committed:
merged with master

liorzblrn committed:
feat: add notification manager

liorzblrn committed:
feat: webhook manager

liorzblrn committed:
feat: add swagger json and yaml

alonp99 committed:
chore(swagger): fix endpoints

alonp99 committed:
merged with master

alonp99 committed:
Merge branch 'dev' into alonp/feat/tm

alonp99 committed:
fix(kyb): logs and kyb header fix

alonp99 committed:
fix(kyb): logs and kyb header fix

liorzblrn committed:
feat: alerts

liorzblrn committed:
feat: alerts

liorzblrn committed:
feat: pagination

liorzblrn committed:
feat: faker

alonp99 committed:
tm model change

liorzblrn committed:
feat: faker alerts

liorzblrn committed:
feat: apply filters

liorzblrn committed:
fix: swagger

alonp99 committed:
tm model change

liorzblrn committed:
fix: swagger

alonp99 committed:
tm model change

alonp99 committed:
fixed mapping

alonp99 committed:
fixed mapping

alonp99 committed:
fixed mapping

alonp99 committed:
fixed mapping

alonp99 committed:
fixed mapping

alonp99 committed:
feat(tm-endpoint): paginiation

liorzam commented on services/workflows-service/src/transaction/transaction.service.ts:
```diff
@@ -0,0 +1,106 @@
+        transactionCurrency: '',
+      };
+
+      await sleep(200);
```
Why do you add sleep?

liorzam committed:
bal 1379 add severity to alert v2 (#2064)

Omri-Levy committed:
WIP - Backoffice alerts view (#2070)

* feat: added initial markup for transaction monitoring alerts

* refactor(backoffice-v2): made the alerts view look closer to Figma design

* feat: alerts

* feat: alerts

* feat: pagination

* feat: faker

* feat: faker alerts

* feat: apply filters

* fix: swagger

* fix: swagger

* refactor(backoffice-v2): addressed sign off comments

* feat(backoffice-v2): added fetchers and schemas for alerts

* feat: added alerts analysis sheet & table

* feat(backoffice-v2): added fallback states for the alerts table

* feat(backoffice-v2): added opening sheet on row click

* feat(backoffice-v2): added sorting

* fix(blocks-package): omri-levy/fix/blocks-types

* feat(backoffice-v2): added pagination for the alerts table

* feat(backoffice-v2): implemented search client-side (not api yet)

* feat(backoffice-v2): added multi select filtering

* refactor(backoffice-v2): commented out alerts search

* refactor(backoffice-v2): moved useTransactionsLogic to a file

* refactor(backoffice-v2): renamed props interface

---------

Co-authored-by: Illia Rudniev <cheskmr@gmail.com>
Co-authored-by: Lior Zamir <liorz@ballerine.com>
Co-authored-by: Alon Peretz <Alonp99@gmail.com>

liorzam committed:
bal 1379 assign alert bulk operation (#2066)

liorzam committed:
bal 1379 decision alert bulk operation (#2067)

alonp99 committed:
draft1

alonp99 committed:
draft1

alonp99 committed:
draft1

alonp99 committed:
draft1

alonp99 committed:
draft1

alonp99 committed:
draft1

alonp99 committed:
draft1

Omri-Levy committed:
WIP - Alerts assignment (#2074)

* feat(backoffice-v2): added ability to select rows

* feat(backoffice-v2): now disabling assign button if no alerts are selected

* feat(backoffice-v2): added a generic dropdown component

* feat(*): added the ability to bulk assign and unassign

* fix(workflows-service): fixed wrong user type

* revert(workflows-service): reverted code changed by linting

* refactor(backoffice-v2): updating naming for pr comments

alonp99 committed:
draft1

alonp99 committed:
draft1

alonp99 committed:
draft1

Omri-Levy committed:
WIP - Alerts decision (#2076)

* feat(*): wip - alerts decision changes

* feat(*): added the functionality of making a decision

* feat(workflows-service): now assigning alerts on decision

* fix(backoffice-v2): coming soon decisions are now disabled

* refactor(backoffice-v2): revert decision now moves an alert to new/triggered

* refactor(backoffice-v2): renamed toast function

* style(schema.prisma): fixed format

* refactor(backoffice-v2): removed false positive mapping

* fix(backoffice-v2): fixed revert decision toast

* refactor(backoffice-v2): addressed PR comments

* refactor(backoffice-v2): addressed PR comments for toasts string literals

Omri-Levy committed:
Backoffice Sidebar hover states (#2087)

* feat(backoffice-v2): added hover styling to the navbar

* feat(backoffice-v2): improved navbar styling

* fix(backoffice-v2): styling for nav items with no children fixes

Omri-Levy committed:
Alerts view transactions drawer (#2086)

* fix(backoffice-v2): fixed wrong parentheses

* feat(backoffice-v2): updated polymorphic component to allow ref and now fetching transactions

* feat(*): added transaction direction

* feat(*): connected business to the alerts drawer column

* feat(*): now displaying the merchant column

* refactor(*): moved alerts drawer to a route of its own

* refactor(backoffice-v2): removed redundant commented out line

* fix(backoffice-v2): fixed merge conflict

* refactor(backoffice-v2): improved value naming

* refactor(backoffice-v2): moved useNavigateBack to a file of its own

* feat(*): added pageSize 50 to transactions and fixed transaction pagination take/size

* refactor(*): addressed PR comments

Omri-Levy committed:
Added transaction currency (#2091)

* feat(backoffice-v2): added currency to transaction amount

* feat(backoffice-v2): added handling for undefined transaction amount

Omri-Levy committed:
Added alerts table empty state (#2093)

* feat(backoffice-v2): added empty state for alerts table

* refactor(backoffice-v2): added type to the NoAlerts component

* fix(backoffice-v2): fixed navbar styling

* fix(backoffice-v2): changed unintentional import with src in its path

alonp99 committed:
feat(tld-progress): progress endpoint addtions

alonp99 committed:
feat(tld-progress): progress endpoint addtions

liorzam committed:
Bal 1427 schema changes (#2100)

* fix: db schema

* fix: db schema

* fix: db schema changes

* fix: db schema changes

---------

Co-authored-by: Lior Zamir <liorz@ballerine.com>

alonp99 committed:
feat(tld-progress): progress endpoint addtions

alonp99 committed:
feat(tld-progress): progress endpoint addtions

alonp99 committed:
feat(tld-progress): progress endpoint addtions

alonp99 committed:
feat(tld-progress): progress endpoint addtions

alonp99 committed:
fix(blocks-package): omri-levy/fix/blocks-types

alonp99 committed:
fix(blocks-package): omri-levy/fix/blocks-types

alonp99 committed:
fix(migration): fix

liorzblrn committed:
fix: schema changes

liorzblrn committed:
fix: schema changes

alonp99 committed:
feat(tld-progress): progress endpoint addtions

liorzblrn committed:
fix: seed

alonp99 committed:
feat(tld-progress): progress endpoint addtions

alonp99 committed:
test fix

alonp99 committed:
test fix

alonp99 committed:
test fix

alonp99 committed:
test fix

alonp99 committed:
test fix

alonp99 committed:
test fix

alonp99 committed:
test fix

alonp99 committed:
test fix

liorzam committed:
fix: swagger (#2101)

* fix: swagger

* fix: revert

* fix: eslint

---------

Co-authored-by: Lior Zamir <liorz@ballerine.com>

Omri-Levy committed:
Alerts drawer heading and summary (#2099)

* feat(*): the transactions drawer no longer uses hardcoded heading and summary

* feat(backoffice-v2): added counterparty id to search query params

* refactor(backoffice-v2): removed alert-definitions domain and moved logic to alerts domain

* feat(backoffice-v2): added loading state to heading and summary

* feat(backoffice-v2): added fallback state to alerts drawer

* refactor(seed.ts): updated seed to alternative between business and counterparty id

* revert(*): reverted counterparty id changes

* fix(*): fixed changes that broke from conflicts

* refactor(workflows-service): added optional chaining

* fix(workflows-service): lint error

Omri-Levy committed:
Fixed new status applying on decision (#2105)

* fix(backoffice-v2): fixed filters in state but not in the url reaching the backend

* refactor(backoffice-v2): removed console.log

* revert(backoffice-v2): removed assignee default filter

* fix(backoffice-v2): fixed key warning on navbar

* fix(*): lockfile fix

alonp99 committed:
Bal 1427 alert definition v3 (#2107)

alonp99 committed:
Small model/seed changes (#2106)

Omri-Levy committed:
Updated instances of alert state (#2109)

* refactor(*): updated instances of alert state

* refactor(backoffice-v2): updated toast text

* refactor(backoffice-v2): simplified pattern matching for decision

* refactor(workflows-service): addressed pr comment

* fix(workflows-service): fixed return of function

liorzblrn committed:
chore: reset datamigration sub module ref

Omri-Levy committed:
chore(*): merged dev to feature branch

alonp99 committed:
Merge branch 'dev' into alonp/feat/tm

liorzam committed:
Bal 1405 check alert def service (#2132)

MatanYadaev committed:
fix(get-alerts-search-schema.ts): add alertType field to the search schema to support filtering by alert type
fix(TransactionMonitoringAlertsAnalysis.page.tsx): add key prop to AlertAnalysisSheet component to force re-render when alertId changes
fix(useTransactionMonitoringAlertsAnalysisPageLogic.tsx): remove duplicate navigateBack() call in useEffect hook
fix(schema.prisma): fix indentation and rename underReview enum value to under_review for consistency
fix(alert.service.ts): fix mapping of under_review enum value to pending status
